### ISSUE TITLE
[Bug] - Cassendra Timeout Issue

### ISSUE THREAD

**Issue by pankajjangid05** [samagra-comms/orchestrator#41]
**Getting Exception**

- [x] Error in getLatestXMessageReactivePreparedStatementCallback; CQL [SELECT * FROM xmessage WHERE userid=? AND timestamp>? ALLOW FILTERING]; Query timed out after PT2S; nested exception is com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S

**Comment by pankajjangid05** [comment]
Trying to regenerate this error on my local or staging.

**Comment by karntrehan** [comment]
Testing undertaken yesterday. Cassandra still broke. Checking config issues.

**Comment by karntrehan** [comment]
Prod casandra is a 4 core, 8 GB ram. This could be a problem. 
Pre prod is a 8 core, 16 gb ram system. We will try triggering 70K notification using it and config config is an issue.

**Comment by pankajjangid05** [comment]
This error is not generated again on prod.

**Comment by pankajjangid05** [comment]
Yesterday we get some error related to Cassandra but it's not related to the notification. 

Error in getLastMessageIDReactivePreparedStatementCallback; CQL [SELECT * FROM xmessage WHERE userid=? AND timestamp>? ALLOW FILTERING]; Query timed out after PT2S; nested exception is com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S

**Comment by geeky-abhishek** [comment]
@pankajjangid05 will pick this after notifications task

**Comment by karntrehan** [comment]
To check if we want to increase timeout or debug further on issue.

**Comment by pankajjangid05** [comment]
**Issue Figure out**

During the recent tests, I encountered an error when performing parallel processes involving the insertion and selection of data from Cassandra. Here are a few real examples:

1. Last week, I sent 100,005 notifications from the staging server. After initiating this process, the outbound starts the insertion of records into Cassandra. To cross-check, I parallel-ran a select query in the Cassandra container. As a result, the outbound insertion was interrupted due to the select query. Once the select query is completed, the outbound insertion code resumed its operation.
2. Following the previous test, I triggered notifications again for 100,005 users. While running this process, I encountered a "Query timed out after PT2S" error in the Orchestrator service. This error occurred because some other users had initiated conversations with the bot. When starting a conversation, we retrieve the messageId from Cassandra using the following query: "SELECT * FROM xmessage WHERE userid=? AND timestamp>? ALLOW FILTERING". As the insertion process was executed, Cassandra did not respond in a timely manner, so this error comes.

**Tests**

To resolve this issue I increase the Cassandra "basic.request.timeout" time by 30 seconds in orchestrator service and request 100 request parallel to getMessageId using this query "SELECT * FROM xmessage WHERE userid=? AND timestamp>? ALLOW FILTERING", but Cassandra gives some new error which is "Caused by: com.datastax.oss.driver.api.core.servererrors.ReadFailureException: Cassandra failure during read query at consistency LOCAL_ONE (1 responses were required but only 0 replica responded, 1 failed)".
